spring-kafka/src/main/java/org/springframework/kafka/listener/DeadLetterPublishingRecoverer.java

```diff
@@ -79,6 +87,14 @@
 
 	private Duration partitionInfoTimeout = Duration.ofSeconds(FIVE);
 
+	private boolean replaceOriginalHeaders = true;
+
+	private boolean failIfSendResultIsError = false;
+
+	private boolean throwIfNoDestinationReturned = false;
```

Comment:
I think both of these should be true by default.

Reply:
Ok. By doing so, many tests break in the DLPR tests because the SettableListenableFuture times out on get, as it has no value. I'll assign some dummy value so it works again.